flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -296,6 +318,13 @@ private void commitOperation(SnapshotUpdate<?> operation, int numDataFiles, int
 
     long start = System.currentTimeMillis();
     operation.commit(); // abort is automatically called if this fails.
+
+    Long watermarkForCheckpoint = watermarkPerCheckpoint.get(checkpointId);
```

Comment:
We need to use table transaction here so that `operation.commit()` and `table.updatePropertie...commit() are atomic.

Reply:
Yes, if we store watermark in snapshot summary metadata as @rdblue said, we can omit `table.updatePropertie` transaction.

Reply:
We actually [set it as table properties too](https://github.com/Netflix-Skunkworks/nfflink-connector-iceberg/blob/master/nfflink-connector-iceberg/src/main/java/com/netflix/spaas/nfflink/connector/iceberg/sink/IcebergCommitter.java#L555).

I think table properties is easier for the workflow scheduler (in the batch system) to query. Otherwise, they have to iterate the snapshots and find out the latest watermarks for all 3 regions. cc @rdblue